core/vm/errors.go

```diff
@@ -34,6 +34,7 @@ var (
 	ErrWriteProtection          = errors.New("write protection")
 	ErrReturnDataOutOfBounds    = errors.New("return data out of bounds")
 	ErrGasUintOverflow          = errors.New("gas uint64 overflow")
+	ErrInvalidCode              = errors.New("invalid code")
```

Comment:
Slightly unrelated question (not blocking this PR): EVMC has a `CONTRACT_VALIDATION_FAILED` error code (used as part of Ewasm 1.0). Probably we could use that error code for EOF1 validation, but should we use it already for this use case, or it makes more sense separating the error codes?